R/PredictionSurv.R

```diff
@@ -171,10 +171,12 @@ PredictionSurv = R6Class("PredictionSurv",
       }
     },
     .distrify_survarray = function(x) {
-      if (inherits(x, "array")) { # can be matrix as well
+      if (inherits(x, "array") && nrow(x) > 0) { # can be matrix as well
         # create Matdist or Arrdist (default => median curve)
         distr6::as.Distribution(1 - x, fun = "cdf",
           decorators = c("CoreStatistics", "ExoticStatistics"))
+      } else {
+        NULL
```

Comment:
@bblodfon  What's this case?

Reply:
This is the result of me harmonizing the output of `p$distr` when you filter too much :) See tests [here](https://github.com/mlr-org/mlr3proba/blob/fix_rcll_bottleneck/tests/testthat/test_PredictionSurv.R#L213). 

So if you do `p$filter(42)` and `42` is not a `row_id` you end up with 0 observations. But if `p$data$distr` is a matrix it actually leaves a 0-row matrix with the time points as columnames with the `drop = FALSE` (https://github.com/mlr-org/mlr3proba/blob/fix_rcll_bottleneck/R/PredictionDataSurv.R#L144). On the other hand, if `p$data$distr` is a distribution, it will leave `NULL` with the use of `switch`, which makes sense: https://github.com/mlr-org/mlr3proba/blob/fix_rcll_bottleneck/R/PredictionDataSurv.R#L140.